client/src/general_utils.ts

```diff
@@ -117,17 +137,21 @@ export const retry_promise = (promise_to_try, retries = 2, interval = 500) => {
 // From https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign
 // "This is an assign function that copies full descriptors"
 // Here, used to preserve getters (like d3-selection.event), as opposed to Object.assign which just copies the value once (null for most)
-export function completeAssign(target, ...sources) {
+export function completeAssign(target: any, ...sources: any[]) {
```

Comment:
Took a look at `@types/lodash`'s typing for their assign (slightly different implementation, but should be the same type signatrue). Looks like this
```
        assign<TObject, TSource>(object: TObject, source: TSource): TObject & TSource;
        /**
         * @see _.assign
         */
        assign<TObject, TSource1, TSource2>(object: TObject, source1: TSource1, source2: TSource2): TObject & TSource1 & TSource2;
        /**
         * @see _.assign
         */
        assign<TObject, TSource1, TSource2, TSource3>(object: TObject, source1: TSource1, source2: TSource2, source3: TSource3): TObject & TSource1 & TSource2 & TSource3;
        /**
         * @see _.assign
         */
        assign<TObject, TSource1, TSource2, TSource3, TSource4>(object: TObject, source1: TSource1, source2: TSource2, source3: TSource3, source4: TSource4): TObject & TSource1 & TSource2 & TSource3 & TSource4;
        /**
         * @see _.assign
         */
        assign<TObject>(object: TObject): TObject;
        /**
         * @see _.assign
         */
        assign(object: any, ...otherArgs: any[]): any;
```

Reply:
Yeah I think I understand what they're doing but I'm having hard time actually using it. I pushed the code on what I have on this